core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1504,22 +1504,37 @@ class UnifiedLog(@volatile var logStartOffset: Long,
 
   /**
    * Flush all local log segments
+   *
+   * @param inclusive should be true during a clean shutdown, and false otherwise. The reason is that
+   * we have to pass logEngOffset + 1 to the `localLog.flush(offset: Long): Unit` function to flush empty
+   * active segments, which is important to make sure we don't lose the empty index file during shutdown.
    */
-  def flush(): Unit = flush(logEndOffset)
+  def flush(inclusive: Boolean): Unit = flush(logEndOffset, inclusive)
```

Comment:
Can we call this parameter as `forceFlushActiveSegment`? I don't understand what does `inclusive` mean in this public API.